sys/net/gnrc/pktbuf_static/gnrc_pktbuf_static.c

```diff
@@ -174,33 +174,37 @@ int gnrc_pktbuf_realloc_data(gnrc_pktsnip_t *pkt, size_t size)
                           _align(sizeof(_unused_t)) : _align(size);
 
     mutex_lock(&_mutex);
-    assert((pkt != NULL) && (pkt->data != NULL) && _pktbuf_contains(pkt->data));
-    if (size == 0) {
-        DEBUG("pktbuf: size == 0\n");
-        mutex_unlock(&_mutex);
-        return ENOMEM;
-    }
+    assert((pkt != NULL) && ((pkt->data == NULL) || _pktbuf_contains(pkt->data)));
```

Comment:
shouldn't that be an `||` instead of an `&&`? The original precondition is `(pkt->size > 0) => (pkt->data != NULL)`. In [boolean algebra that would be](https://en.wikipedia.org/wiki/Boolean_algebra#Secondary_operations) `(pkt->size == 0) || (pkt->data == NULL)`, so

```C
assert((pkt != NULL) && ((pkt->data == NULL) || (pkt->size == 0) || _pktbuf_contains(pkt->data));
```